packages/perseus-editor/src/components/locked-circle-settings.tsx

```diff
@@ -0,0 +1,214 @@
+import {components} from "@khanacademy/perseus";
+import {View, useUniqueIdWithMock} from "@khanacademy/wonder-blocks-core";
+import {OptionItem, SingleSelect} from "@khanacademy/wonder-blocks-dropdown";
+import {TextField} from "@khanacademy/wonder-blocks-form";
+import {Strut} from "@khanacademy/wonder-blocks-layout";
+import {spacing} from "@khanacademy/wonder-blocks-tokens";
+import {LabelMedium, LabelLarge} from "@khanacademy/wonder-blocks-typography";
+import {StyleSheet} from "aphrodite";
+import * as React from "react";
+
+import CircleSwatch from "./circle-swatch";
+import ColorSelect from "./color-select";
+import CoordinatePairInput from "./coordinate-pair-input";
+import LockedFigureSettingsAccordion from "./locked-figure-settings-accordion";
+import LockedFigureSettingsActions from "./locked-figure-settings-actions";
+
+import type {AccordionProps} from "./locked-figure-settings";
+import type {
+    LockedCircleFillType,
+    LockedCircleType,
+} from "@khanacademy/perseus";
+
+const {InfoTip} = components;
+
+export type Props = AccordionProps &
+    LockedCircleType & {
+        /**
+         * Called when the delete button is pressed.
+         */
+        onRemove: () => void;
+        /**
+         * Called when the props (coords, color, etc.) are updated.
+         */
+        onChangeProps: (newProps: Partial<LockedCircleType>) => void;
+    };
+
+const LockedCircleSettings = (props: Props) => {
+    const {
+        center,
+        radius,
+        color,
+        fillStyle,
+        strokeStyle,
+        expanded,
+        onToggle,
+        onChangeProps,
+        onRemove,
+    } = props;
+
+    const [radiusInput, setRadiusInput] = React.useState(radius.toString());
+
+    const ids = useUniqueIdWithMock();
+    const radiusInputId = ids.get("radius-input");
+    const strokeSelectId = ids.get("stroke-style-select");
+    const fillSelectId = ids.get("fill-style-select");
+
+    function handleCenterChange(newCenter) {
+        onChangeProps({center: newCenter.coord});
+    }
+
+    function handleRadiusChange(newValue) {
+        // Update the local state (update the input field value).
+        setRadiusInput(newValue);
+
+        // If the new value is not a number, don't update the props.
+        // If it's empty, keep the props the same value instead of setting to 0.
+        if (isNaN(+newValue) || newValue === "") {
```

Comment:
I did it this way because the number input sets the value to 0 if the input is not a number. That means if you have a circle with a radius "1" and then you erase it in preparation to type something else, or if you type a "." before typing ".5" it makes the circle disappear. Personally, I like the previous state to remain instead of the circle disappear before you're done typing. I can remove this line if you don't like that though. I'll update the comment for clarity in the meantime.